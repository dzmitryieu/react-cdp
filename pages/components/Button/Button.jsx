import React from 'react';
import PropTypes from 'prop-types';
import styles from './Button.scss';

const Button = ({
  id,
  type,
  text,
  onClick,
  isMain,
  className,
  dataAttr,
}) => (
  <button
    id={id}
    className={`${styles.button} ${isMain ? styles.big : ''} ${styles[type]} ${className}`}
    type="button"
    data-attr={dataAttr}
    onClick={onClick}
  >
    {text}
  </button>
);

Button.propTypes = {
  type: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  onClick: PropTypes.func,
  isMain: PropTypes.bool,
  className: PropTypes.string,
  id: PropTypes.string,
  dataAttr: PropTypes.string,
};
Button.defaultProps = {
  onClick: null,
  isMain: null,
  className: null,
  id: null,
  dataAttr: null,
};
export default Button;
