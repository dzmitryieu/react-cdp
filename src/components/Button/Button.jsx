import React from 'react';
import PropTypes from 'prop-types';
import styles from './Button.scss';

const Button = ({
  id,
  type,
  text,
  onClick,
  isMain,
  className,
}) => (
  <button id={id} className={`${styles.button} ${isMain ? styles.big : ''} ${styles[type]} ${className}`} type="button" onClick={onClick}>
    {text}
  </button>
);

Button.propTypes = {
  type: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  onClick: PropTypes.func,
  isMain: PropTypes.bool,
  className: PropTypes.string,
  id: PropTypes.string,
};
Button.defaultProps = {
  onClick: null,
  isMain: null,
  className: null,
  id: null,
};
export default Button;
