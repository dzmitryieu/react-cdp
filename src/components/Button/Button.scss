@import "../../utils/constants.scss";

.button {
  height: 30px;
  width: 100px;
  color: white;
  text-align: center;
  font-size: 14px;
  font-family: $helvetica;
  line-height: 22px;
  cursor: pointer;
  transition: all 0.3s;
  margin: 10px;
  outline: none;
  text-transform: uppercase;
  border: none;

  &:hover {
    cursor: pointer;
  }
}

.big {
  width: 150px;
  height: 40px;
  margin: 0;
}

.red {
  background-color: $light-red;

  &:hover {
    background-color: lighten($light-red, 20%);
  }
}

.black {
  background-color: $dark-gray;

  &:hover {
    color: darken(white, 20%);
  }
}
