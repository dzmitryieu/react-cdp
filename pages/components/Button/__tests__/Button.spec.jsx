import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Button from '../index';

const renderer = new ShallowRenderer();
const data = {
  type: 'black',
  text: 'View Info',
  onClick: () => ('Hello!'),
};

const blueData = {
  type: 'blue',
  text: 'View Info',
  onClick: () => ('Hello!'),
};

const mainButtonData = {
  type: 'blue',
  text: 'View Info',
  isMain: true,
  onClick: () => ('Hello!'),
};

describe('Button Snapshot', () => {
  test('renders', () => {
    const component = renderer.render(
      <Button {...data} />,
    );
    expect(component).toMatchSnapshot();
  });
});

describe('Button blue Snapshot', () => {
  test('renders', () => {
    const component = renderer.render(
      <Button {...blueData} />,
    );
    expect(component).toMatchSnapshot();
  });
});

describe('Button blue Snapshot', () => {
  test('renders', () => {
    const component = renderer.render(
      <Button {...mainButtonData} />,
    );
    expect(component).toMatchSnapshot();
  });
});
