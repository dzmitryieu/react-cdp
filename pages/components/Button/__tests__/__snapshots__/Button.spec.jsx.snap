// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Snapshot renders 1`] = `
<button
  className="button  black null"
  data-attr={null}
  id={null}
  onClick={[Function]}
  type="button"
>
  View Info
</button>
`;

exports[`Button blue Snapshot renders 1`] = `
<button
  className="button  blue null"
  data-attr={null}
  id={null}
  onClick={[Function]}
  type="button"
>
  View Info
</button>
`;

exports[`Button blue Snapshot renders 2`] = `
<button
  className="button big blue null"
  data-attr={null}
  id={null}
  onClick={[Function]}
  type="button"
>
  View Info
</button>
`;
